### PR TITLE
feat: Add user_id in webhook migration

### PR DESCRIPTION
**Pull request by jonas-martinez**
See Epic https://github.com/lenra-io/Lenra/issues/141

## Checklist
- [x] I didn't over-scope my PR
- [x] My PR title matches the [commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I did not include breaking changes
- [x] I included unit tests that cover my changes
- [ ] I added/updated the documentation about my changes
- [ ] I made my own code-review before requesting one

## Description of the changes

We needed to make it possible for a webhook to be specific to a user instead of an environment, this could be useful if a webhook only needs to target a specific user and not trigger listeners outside of this user.

### CODE REVIEW

File: lib/webhooks/webhooks.ex [hunk 1]
```diff
@@ -23,12 +24,18 @@ defmodule ApplicationRunner.Webhooks.Webhook do
...
     |> foreign_key_constraint(:environment_id)
+    |> foreign_key_constraint(:user_id)
   end
 
   def new(env_id, params) do
     %Webhook{environment_id: env_id}
     |> Webhook.changeset(params)
   end
+
+  def new(env_id, user_id, params) do
```
Comment (jonas-martinez) [inline lib/webhooks/webhooks.ex]: @pichoemr Is it better to have two `new` methods ? One with the env_id and one with the env_id + user_id ?

Or is it better to have only one `new` method only with the required `env_id` parameter and pass the `user_id` in the `params` (and cast it in the changeset) ?

**Review by shiipou — APPROVED** [review]

**Review by pichoemr — COMMENTED** [review]
Can you add a test to get the webhook by user preloaded?

**Review by pichoemr — COMMENTED** [review]
You have a warning on line 57 of webhook_test, 
replace `[action: reason]` by `[action: _reason]`

**Pull request merged**